ios-offline-maps/ios-offline-maps/TileRegionDownloadView.swift

```diff
@@ -0,0 +1,252 @@
+    
+    var body: some View {
+        NavigationView {
+            VStack {
+                // Header
+                HStack {
+                    Button(action: {
+                        dismiss()
+                    }) {
+                        HStack {
+                            Image(systemName: "chevron.left")
+                            Text("Back")
+                        }
+                        .foregroundColor(.blue)
+                    }
+                    Spacer()
+                    Text("Offline Regions")
+                        .font(.title2)
+                        .fontWeight(.semibold)
+                    Spacer()
+                    Button(action: {
+                        clearAllRegions()
+                    }) {
+                        Text("Clear All")
+                            .font(.caption)
+                            .foregroundColor(.red)
+                    }
+                }
+                .padding()
+                
+                // List of regions
+                List {
+                    ForEach(regions, id: \.id) { region in
+                        RegionRowView(
+                            regionId: region.id,
+                            regionName: region.name,
+                            isDownloading: downloadingRegions.contains(region.id),
+                            progress: downloadProgress[region.id] ?? 0.0,
+                            refreshTrigger: refreshTrigger,
+                            onDownload: {
+                                downloadRegion(region: region)
+                            }
+                        )
+                    }
+                }
+            }
+        }
+        .navigationBarHidden(true)
+    }
+    
+    
+    private func downloadRegion(region: OfflineRegion) {
+        OfflineRegionManager.downloadRegion(
+            region: region,
+            downloadingRegions: downloadingRegions,
+            onDownloadingRegionsUpdate: { updatedSet in
+                DispatchQueue.main.async {
+                    self.downloadingRegions = updatedSet
+                }
+            },
+            onProgress: { regionId, progress in
+                DispatchQueue.main.async {
+                    self.downloadProgress[regionId] = progress
+                }
+            },
+            onCompletion: { regionId, result in
+                DispatchQueue.main.async {
+                    // Remove progress tracking since download is complete
+                    self.downloadProgress.removeValue(forKey: regionId)
+                    
+                    switch result {
+                    case .success(let tileRegion):
+                        print("Downloaded \(region.name): \(tileRegion.completedResourceSize) bytes")
+                    case .failure(let error):
+                        print("Failed to download \(region.name): \(error)")
+                    }
+                }
+            }
+        )
+    }
```

Comment:
This view seems to be quite busy managing the state of the downloads, I suggest offloading some of this logic to a view model, similarly how a list of alerts managed here https://toyboy2.medium.com/mvvm-list-with-swiftui-a91425369b67
This way views won't have to manage state themselves, but just bind their state for the state of view model and the states won't have to be passed to the offline region manager as the view model could be the central piece of state management.